### DISCUSSION TITLE
Godot by Example, Godot Cookbook

### DISCUSSION THREAD

**Discussion by wdv4758h** [godotengine/godot-proposals#5655] [General Discussions]
I've been learning Godot recently. The documentation and demo projects repo help a lot. But sometime I found searching the specific game action on YouTube is better for learning the flow. And then go back to read official docs to dive deeper.

And for some programming languages, there is `<LANG> By Example` or `<LANG> Cookbook`, which may have specific subject with quick snippets.

Some samples:
* [Rust by Example](https://doc.rust-lang.org/rust-by-example/index.html#rust-by-example)
* [Rust Cookbook](https://rust-lang-nursery.github.io/rust-cookbook/)
* [Go by Example](https://gobyexample.com/)

During my learning progress, I also take couple few notes about something like this (although not much). For example, if a new dev wants player to pick up an item nearby, what's the available ways? Probably create something like this will help people to learn.

A collection of examples of how to achieve specific actions in game.

**Comment by Sythelux** [comment]
Aren't there currently at least "learn Godot" websites?

https://www.learngodot.com
https://www.gdquest.com
https://godotlearn.com
https://godottutorials.com

I'd say that is a good start.

**Reply by wdv4758h** [reply]
Thanks for the links.

GDQuest did show up from time to time. Others somehow didn't pop up in my search result.

* GDQuest -> I did check it from time to time. Many content. Learned a lot.
* Learn Godot -> 2 course with small project, targeting no audience without programming background. Sounds good. But I haven't paid to see the details, no real comments.
* Godot Learn -> I see 4 articles on the site. Good content.
* Godot Tutorials -> seems to have many things, will check more.

One more adds. The site I found quite good is actually [KidsCanCode's Godot Recipes](https://kidscancode.org/godot_recipes/). They layout things in small targets. And each article defines problem and solution. So I can work on my project and easily come back to pick up one or two thing.
